src/tests/class_tests/openms/source/EmpiricalFormula_test.cpp

```diff
@@ -366,56 +366,87 @@ END_SECTION
 START_SECTION(IsotopeDistribution getIsotopeDistribution(UInt max_depth) const)
   EmpiricalFormula ef("C");
   IsotopeDistribution iso = ef.getIsotopeDistribution(CoarseIsotopePatternGenerator(20));
-  double result[] = { 0.9893, 0.0107};
+  double result_mass[] = { 12, 13.0033548378 };
+  double result_prob[] = { 0.9893, 0.0107 };
   Size i = 0;
   for (IsotopeDistribution::ConstIterator it = iso.begin(); it != iso.end(); ++it, ++i)
   {
-    TEST_REAL_SIMILAR(it->getIntensity(), result[i])
+    TEST_REAL_SIMILAR(it->getIntensity(), result_prob[i])
+    // accurate masses should have been calculated
+    TEST_REAL_SIMILAR(it->getMZ(), result_mass[i])
   }
+
+  IsotopeDistribution iso2 = ef.getIsotopeDistribution(CoarseIsotopePatternGenerator(20, true));
+  double result_rounded_mass[] = { 12, 13 };
+  i = 0;
+  for (IsotopeDistribution::ConstIterator it = iso2.begin(); it != iso2.end(); ++it, ++i)
+  {
+    // probabilities should not change due to mass vs atomic number calculations
+    TEST_REAL_SIMILAR(it->getIntensity(), result_prob[i])
+    // atomic numbers should have been calculated
+    TEST_REAL_SIMILAR(it->getMZ(), result_rounded_mass[i])
+  }
+
+  // Formula of Bovine Serum Albumin (BSA) that is has been processed and is missing the first 24 AA
+  EmpiricalFormula ef2("C2934H4615N781O897S39");
+  IsotopeDistribution iso3 = ef2.getIsotopeDistribution(CoarseIsotopePatternGenerator(20));
+  // monoisotopic mass
+  TEST_REAL_SIMILAR(iso3.begin()->getMZ(), 66389.863)
```

Comment:
@DennisGoldfarb once you are here, why not also test the probability? Lets also test one of the more interesting ones, eg 20 or 30 isotopes in to see whether we get the correct result